crypto/dsa/ecdsa/ecdsa.go

```diff
@@ -0,0 +1,83 @@
+package ecdsa
+
+import (
+	"fmt"
+
+	"github.com/tbd54566975/web5-go/jwk"
+)
+
+const (
+	KeyType = "EC"
+)
+
+var AlgorithmIDs = map[string]bool{
+	SECP256K1AlgorithmID: true,
+}
+
+func GeneratePrivateKey(algorithmID string) (jwk.JWK, error) {
+	var privateKey jwk.JWK
+	var err error
+
+	switch algorithmID {
+	case SECP256K1AlgorithmID:
+		privateKey, err = SECP256K1GeneratePrivateKey()
+	default:
+		err = fmt.Errorf("unsupported algorithm: %s", algorithmID)
+	}
+
+	return privateKey, err
+}
+
+func GetPublicKey(privateKey jwk.JWK) jwk.JWK {
```

Comment:
There seems to be a common interface to these packages, though I'm not 100% sure. If there is I'd probably create an interface that can be reused.

Reply:
@alecthomas I recall @mistermoe mentioning that you were critical of the usage of interfaces in Golang. Which circumstances are a good usage of interfaces and which circumstances are a bad usage of interfaces?

Reply:
The rule is generally "don't leap to using interfaces by default". This is quite different from languages where interfaces are declared on the concrete types directly, and you're often forced to use interfaces in order to fake those implementations out for testing. Basically in Go you often flip how interfaces traditionally work, from declaring them as part of the implementation API, to declaring them where they're used.

If you have common behaviour across a bunch of implementations, and a need to access that behaviour in a polymorphic manner, then it probably makes sense to use them in a more traditional manner though. This seems like one of those cases, though again, I could be wrong.